src/basic/cgroup-util.h

```diff
@@ -183,9 +183,13 @@ typedef enum CGroupUnified {
 int cg_path_open(const char *controller, const char *path);
 int cg_cgroupid_open(int fsfd, uint64_t id);
 
+typedef enum CGroupPidFlags {
+        CGROUP_PID_RETURN_UNMAPPED = 1 << 0,
```

Comment:
Please add a short comment about the flag.

Also, I'd like to name the flag more explicit, e.g. CGROUP_PID_REFUSE_UNMAPPED or CGROUP_PID_REFUSE_ZERO. Using 'RETURN' is not clear if `cg_read_pid()` will succeed or not.

Reply:
Hm, those names seem opposite to what the flag is doing.
Since the function will not refuse them when the flag is set, but return them instead of skipping them.

The default is to skip the zero/unmapped pids, and the flag makes the function return the zero ones instead, so the caller can decide what to do with them. It will never not succeed due to the pid being zero.

Maybe _PROCESS_UNMAPPED? But that seems less clear than RETURN to me. Since when the flag is set, the function will return the unmapped pids, and it won't when it's unset.

Reply:
Ah, OK, then how about CGROUP_IGNORE_ZERO_PID or so?

Reply:
Would need to be DONT_IGNORE_ZERO_PID then.

Reply:
Huh, I was confused. So, re-reading the code and your comment in the above, then I now prefer REFUSE_ZERO, or something explicit that reading zero goes into failure.

PROCESS_UNMAPPED is not clear if we refuse or skip zeros for me.
We usually not use negative naming, so DONT_IGNORE is not good.

Reply:
Yeah, DONT_IGNORE_ZEROS is definitely ugly and clunky.
Though I still feel like REFUSE_ZERO is the wrong way around. Since the flag does not make it refuse anything. Quite the opposite, it makes it not skip the zeros, and return them as-is instead.

And the term "refuse" also makes me think it'd throw an error, but it does not throw an error in either case.

The flag toggles between the function skipping over zeros(which is the flags==0 default case), or returning them like any other pid when the flag is set.
So the flag saying SKIP or REFUSE seems wrong, since that's the exact opposite of what it does.

How about CGROUP_PID_INCLUDE_UNMAPPED/ZERO?